### ISSUE TITLE
Site mobile menu is missing 

### ISSUE THREAD

**Issue by zdrawku** [IgniteUI/igniteui-angular-samples#905]
### Description  
Open any topic on mobile view and click on the main site menu. It is not populated with data.

### Steps to reproduce  

1. Open this [topic](https://www.infragistics.com/products/ignite-ui-angular/angular/components/carousel.html)
2. Click on the hamburger menu
3. Empty list

### Result  
![image](https://user-images.githubusercontent.com/11193764/51677926-c5ea1380-1fe3-11e9-9eec-d4c920a7fa86.png)

### Expected result
![image](https://user-images.githubusercontent.com/11193764/51677994-f2059480-1fe3-11e9-8c34-2960fa710c64.png)

**Comment by HristoP96** [comment]
@jasonzajac , could you please take a look at this issue?

After investigating what may trigger such behavior for mobile versions, it appears that the classes, which are applied to the `globalnav` element, are controlled by this script: `"https://infragistics.com/assets/modern/scripts/navigation.js"`, which is contained in the [igniteui-docfx-template](https://github.com/IgniteUI/igniteui-docfx-template/blob/master/template/partials/scripts.tmpl.partial).

I made a copy of this file, referenced it locally  and I saw a strange performance, triggered by these lines:

![navigation](https://user-images.githubusercontent.com/25771040/51824105-0e5a4780-22ea-11e9-9300-df0a8949d6f8.PNG) 

![ezgif com-video-to-gif 5](https://user-images.githubusercontent.com/25771040/51827610-7f9df880-22f2-11e9-9dbc-dab749314837.gif)


On one click, the dropdown opens and then immediately closes itself.

Removing the above shown code lines, the behavior is no more reproduced:

![ezgif com-video-to-gif 4](https://user-images.githubusercontent.com/25771040/51826811-980d1380-22f0-11e9-96ba-e1453b1ac038.gif)

**Comment by jasonzajac** [comment]
Thanks @HristoP96 for spotting this! I think it might be due to the fallback HTML making use of the auto init data attribute (set to true instead of false). Making that small changed resolved the issue when testing locally.

@zdrawku I've submitted a PR, https://github.com/IgniteUI/igniteui-docfx-template/pull/77

BTW, I noticed two other bugs while working on this:
1. When the global search is revealed, the search submit button doesn't have an icon. It appears that a CSS rule within the application's stylesheet is setting it to none. I've included a fix for this in the PR above.
![image](https://user-images.githubusercontent.com/1648846/52353722-1b1f1f80-29fd-11e9-8b6e-9038153c9d38.png)
2. Mobile Menu - when the users expands up a few different menu items in the mobile menu, they are unable to access items out of the viewport. In the other web properties, we allow them to scroll the whole menu (see [WPF Help](https://www.infragistics.com/help/wpf/)); however, because the header is fixed the user is unable to scroll. I didn't want to take any liberties on making decisions with resolving this issue, since it should require input from your team. So, I thought I'd address it with you guys first. We can create a new bug around this issue and determine best path forward.
